all-rit/ALL
# 573 Story to Create Quiz Service

# 573 Story to Create Quiz Service
closes #573 

## Description 
This story is intended on finalizing the ability to making the quiz database driven and not dependent on the API directory., eliminating the need for extra files storing json within the codebase

### User Story
- As a **Developer** I would like to **have the quiz database driven** so **we can update a single source of truth**

- [x] No discrepancies across browsers. Eg. chrome vs safari
- [x] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [x] CSS (https://jigsaw.w3.org/css-validator/)
- [x] Google sign-in is operational
- [x] Accessibility functions
- [x] Pages can scale without distorting page
- [x] No dead links
- [x] Navbar is consistent across the site
- [x] Pages are screen-reader accessible
- [x] Contrast meets standards for accessibility
- [x] Pages are keyboard accessible
- [x] Passes WAVE Evaluation Tool chrome extension


### Testing instructions 

1. cd into database directory and drop the db container using `` docker compose down -v``
2. bring back up with ``docker compose up``
3. cd into the server directory and run ``npm test`` and ensure tests pass 
4. then use pm2 or manually run the setup 
5. go through the quizzes on the site
